### PR TITLE
Respect .prettierignore config

### PR DESCRIPTION
**Pull request by kapowaz**
This PR fixes an issue with eslint-plugin-prettier not respecting file ignore patterns specified in `.prettierignore` (#88).

### CODE REVIEW

File: eslint-plugin-prettier.js [hunk 1]
```diff
@@ -401,48 +402,52 @@ module.exports = {
...
-                      difference.deleteText,
-                      difference.insertText
-                    );
-                    break;
-                }
-              });
-            }
+
+            prettier.getFileInfo(sourceFileName, {
+              ignorePath: '.prettierignore'
```
Comment (kapowaz) [inline eslint-plugin-prettier.js]: I’m not sure if there’s some generic mechanism for exposing default config so that people can specify `eslint-plugin-prettier` to use a different file than `.prettierignore`, but given that this was broken until now I’m not sure just how much it matters… maybe one to be addressed in a later PR? 🤷‍♂️

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v2.6.3 (2018-07-17)
+
+* Fix: respect .prettierignore settings ([#88](https://github.com/prettier/eslint-plugin-prettier/issues/88))
+
```
Comment (not-an-aardvark) [inline CHANGELOG.md]: It's not necessary to update the changelog -- it will be automatically updated during the release.

File: eslint-plugin-prettier.js [hunk 3]
```diff
@@ -401,48 +402,55 @@ module.exports = {
...
+            }).then((sourceFileInfo) => {
```
Comment (not-an-aardvark) [inline eslint-plugin-prettier.js]: I don't think this will work -- ESLint rules are required to be synchronous. Does prettier provide a synchronous API for this?

File: CHANGELOG.md (hunk 2 above)
Comment (kapowaz) [reply CHANGELOG.md]: Great! I did wonder if that might be automated… 🙈

File: eslint-plugin-prettier.js (hunk 3 above)
Comment (kapowaz) [reply eslint-plugin-prettier.js]: Yes! I tried both ways so I can update this to use the `.sync` version 👍

**Comment by kapowaz** [comment]
Thanks for the comments — I’ve pushed a revised PR, hopefully this should address the issues. The tests still fail due to the issues mentioned in #98, but assuming that can be fixed I think this should be good.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-prettier",
-  "version": "2.6.2",
+  "version": "2.6.3",
```
Comment (not-an-aardvark) [inline package.json]: Sorry, I should have mentioned this before, but it's also not necessary to update the version in `package.json` (it will also be automatically updated on release).

File: eslint-plugin-prettier.js [hunk 5]
```diff
@@ -401,19 +402,22 @@ module.exports = {
...
               {},
               prettierRcOptions,
               eslintPrettierOptions,
-              { filepath: context.getFilename() }
+              { filepath: sourceFileName }
             );
-
             const prettierSource = prettier.format(source, prettierOptions);
-            if (source !== prettierSource) {
+            const sourceFileInfo = prettier.getFileInfo.sync(sourceFileName, {
```
Comment (not-an-aardvark) [inline eslint-plugin-prettier.js]: Can you add a test for this behavior?

Comment (not-an-aardvark) [reply eslint-plugin-prettier.js]: Can you check for the existence of `prettier.getFileInfo.sync` before calling it? The plugin needs to continue being able to work for users on old versions of `prettier`.

Comment (kapowaz) [reply eslint-plugin-prettier.js]: I’ll look into adding a test for this.

As regards checking for the existence of `prettier.getFileInfo.sync` I could do this, but this functionality is required in order to test for whether a file is being ignored,  and so I’ve bumped the dependency to prettier `^1.13.0` for that reason. Is there a reason even future releases of `eslint-plugin-prettier` _needs_ to always support old versions of prettier?

Maybe this be a minor release instead of a patch release, perhaps?

Comment (not-an-aardvark) [reply eslint-plugin-prettier.js]: Since the user installs `prettier` independently of `eslint-plugin-prettier`, it would be a breaking change to drop support for old versions of `prettier`, because we don't have any control over what version the user has installed.

Comment (kapowaz) [reply eslint-plugin-prettier.js]: I assume that’s an intentional thing, rather than explicitly having this depend on a given version of prettier? I’m not sure it would be possible to add a test to cover checking for a file being correctly ignored if we can’t guarantee a minimum version of prettier is installed though. Would welcome ideas if you think that is possible!

**Comment by kapowaz** [comment]
I’ve updated the PR again with some existence-guarding checks against `prettier.getFileInfo`. As I commented above, I’m not sure how we can reliably test for this functionality if there is no specific version of prettier guaranteed to be installed (and `getFileInfo` was only added in v1.13.0). Feedback on how to solve that welcomed!

File: eslint-plugin-prettier.js (hunk 5 above)
Comment (not-an-aardvark) [reply eslint-plugin-prettier.js]: Yes, the reasoning is that the user might want to choose a specific version of `prettier` to lint their code with (also see: https://github.com/prettier/eslint-plugin-prettier/pull/1). Admittedly, this was probably more important at the time that pull request was created than it is now, because prettier wasn't yet stable and was making breaking changes more frequently, and people wanted to use the plugin with several different versions.

Comment (kapowaz) [reply eslint-plugin-prettier.js]: Cool. Like I say this can’t be merged until #98 is fixed anyway, so I guess I can revisit when that is handled (or maybe dive in and have a go at fixing that myself! 😁)

Comment (not-an-aardvark) [reply eslint-plugin-prettier.js]: Sure, that would be great.
